elastic/kibana
# [ObsUX] Add UI Setting for controling Profiling visibility in Infra

Closes https://github.com/elastic/kibana/issues/173154

Adds a UI setting to control Infra+Profiling integration from Kibana's Advanced Settings as well as from the Infra Settings screen.

Note that the plugin config feature flag is still there because I realized we need it to disable Profiling integration in serverless.


https://github.com/elastic/kibana/assets/793851/2a5ace9d-9e18-49a4-be95-c722f24072a7

### How to test

* Make sure profiling is enabled in `kibana.dev.yml`
```
xpack.profiling.enabled: true
```
* Start kibana in traditional mode, go to Infra Settings
* Make sure there is the new toggle for Profiling integration and it's on
* Go to one of your host's details and make sure you see the profiling tab
* Toggle the Profiling integration setting off and check that the tap in host details is not visible

* Start kibana in serverless mode
* Make sure there is no new setting neither in Infra Settings nor in Advanced Settings
* Make sure Profiling tab is not visible in host details
